### PR TITLE
dag: add --outs option

### LINKED ISSUE #4532: dvc dag --outs
In the `0.xx` version days, there was a `dvc pipeline show --outs` command that was able to show a DAG that included outputs. 
I thought this was a really nice feature. For instance, I have a `train_test_split` stage that creates two outputs, `train` and `test`, and it would be nice to see them taking different paths in the DAG.
Can you maybe (re)implement this feature?
Thanks!

### PR DESCRIPTION
**Pull request by JIoJIaJIu**
Close #4532

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/command/dag.py [hunk 1]
```diff
@@ -44,8 +44,25 @@ def _build(G, target=None, full=False):
...
         H = G
 
-    def _relabel(stage):
-        return stage.addressing
+    if outs:
+        G = nx.DiGraph()
+        for stage in H.nodes:
+            G.add_nodes_from(stage.outs)
+
+        for from_stage, to_stage in nx.edge_dfs(H):
```
Comment (pared) [inline dvc/command/dag.py:52]: +1 for `edge_dfs`

**Review by pared — APPROVED** [review]

**Review by pmrowla — COMMENTED** [review]
LGTM, I think this probably needs a docs PR to at least update the usage for `dvc dag`?

File: tests/unit/command/test_dag.py [hunk 2]
```diff
@@ -45,11 +45,11 @@ def graph(tmp_dir, dvc):
     return dvc.graph
 
 
-def test_build(graph):
+def test_build(graph):  # pylint: disable=W0621
```
Comment (skshetry) [inline tests/unit/command/test_dag.py:48]: @JIoJIaJIu, you should not get this warning at all, as we use `pylint-pytest` plugin that understands fixture. Could you please check, if all of the dependencies for pylint are installed properly?

**Comment by JIoJIaJIu** [comment]
Based on review  I've removed `pylint: disable=W0621` section and provided update on dvc.org https://github.com/iterative/dvc.org/pull/1918
Please take a look  :eyes: :pray:

**Review by pmrowla — APPROVED** [review]

**Review by skshetry — APPROVED** [review]

**Comment by pmrowla** [comment]
Thanks @JIoJIaJIu 🙏

**Pull request merged**